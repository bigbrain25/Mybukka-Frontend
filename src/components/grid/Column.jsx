import React from 'react';

import PropTypes from 'prop-types';

const Column = ({ classNames, children, ...prop }) => (
  <div className={`col ${classNames}`} {...prop}>{children}</div>
);

export default Column;

Column.defaultProps = {
  classNames: '',
};

Column.propTypes = {
  classNames: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};
