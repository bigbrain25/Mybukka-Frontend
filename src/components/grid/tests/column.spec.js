import React from 'react';

import Column from '../Column';

describe('Column component', () => {
  const wrapper = shallow(<Column classNames="col-4 col-xs-3">value</Column>);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('appends classnames to class col', () => {
    const classNames = wrapper.props().className;
    expect(classNames).toEqual('col col-4 col-xs-3');
  });
});
