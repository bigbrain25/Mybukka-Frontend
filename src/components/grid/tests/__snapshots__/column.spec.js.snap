// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component renders properly 1`] = `ShallowWrapper {}`;
